packages/grafana-ui/src/components/FormField/FormField.tsx

```diff
@@ -1,25 +1,31 @@
 import React, { InputHTMLAttributes, FunctionComponent } from 'react';
+// import React, { InputHTMLAttributes } from 'react';
 import { FormLabel } from '../FormLabel/FormLabel';
 
 export interface Props extends InputHTMLAttributes<HTMLInputElement> {
   label: string;
   labelWidth?: number;
   inputWidth?: number;
+  inputEl?: React.ReactNode;
 }
 
 const defaultProps = {
   labelWidth: 6,
   inputWidth: 12,
 };
 
-const FormField: FunctionComponent<Props> = ({ label, labelWidth, inputWidth, ...inputProps }) => {
+/**
+ * Default form field including label used in Grafana UI. Default input element is simple <input />. You can also pass
+ * custom inputEl if required in which case inputWidth and inputProps are ignored.
+ */
+export const FormField: FunctionComponent<Props> = ({ label, labelWidth, inputWidth, inputEl, ...inputProps }) => {
   return (
     <div className="form-field">
       <FormLabel width={labelWidth}>{label}</FormLabel>
-      <input type="text" className={`gf-form-input width-${inputWidth}`} {...inputProps} />
+      {inputEl || <input type="text" className={`gf-form-input width-${inputWidth}`} {...inputProps} />}
```

Comment:
I think this is an interesting change, I like that we can extend this component with different input elements. Would like to get some feedback from @dprokop @peterholmberg @jschill as well what they think. Another question is should inputEl be a normal React.ReactNode prop, renderProp or children prop?

Reply:
yes @aocenas showed me this and I think it looks sensible. No opinion on prop type though, what's the difference between `ReactNode` and `JSX.Element`.

Reply:
I think using a children instead of `inputEl` would also make sense here. For renderProp I do not think it is needed here because you do not need anything passed from FormField back to the custom input.

> what's the difference between ReactNode and JSX.Element

ReactNode is anything that can be rendered so in addition to components also string, number, null or fragments which as far as I know JSX.Element does not allow.

Reply:
One of the reasons for creating a FormField component is to help the developers align the ui. Allowing a completely custom component adds flexibility but also a greater responsibility on the developer to make sure `inputEl` looks the same everywhere.

That said I dont mean it's a bad thing. But I think a `<FormFieldTextarea />` or `<FormField inputEl={enumWIthFormElements.Textarea}>` is easier to maintain than `<FormField><textarea ..../></FormField>`.  (we could create a `FormFieldTextarea` based on `FormField` and then it's a no-problem)

Reply:
@jschill Yes I think the plan is to have ready to use components like `FormFieldTextarea` and this was done specifically so it is easier to create them (like the `SecretFormField` in this PR), either by us or by users if they have some more advanced use cases. So yes you would go the path of using `inputEl` only if there is no component that suits you already and in that case you are responsible for the alignment and styling.

Reply:
@aocenas Ok! 👍